src/Compilers/CSharp/Portable/BoundTree/Statement.cs

```diff
@@ -445,7 +445,14 @@ internal partial class BoundCatchBlock : ICatchClause
 
         IOperation ICatchClause.Filter => this.ExceptionFilterOpt;
 
-        ILocalSymbol ICatchClause.ExceptionLocal => this.LocalOpt;
+        ILocalSymbol ICatchClause.ExceptionLocal
+        {
+            get
+            {
+                var local = this.Locals.FirstOrDefault();
+                return local?.DeclarationKind == LocalDeclarationKind.CatchVariable ? local : null;
```

Comment:
> This needs to work post-lowering too. Please search through the locals for the exception variable.

There is no need to do the search, if exception local is present, its the first one.

> Also I think ICatchClause needs to be modified to expose all the locals.

That is something to consider, but I am not going to address this right now. Design of IOperation isn't finalized yet.